iot_verifier/src/reward_share.rs

```diff
@@ -237,17 +235,20 @@ impl GatewayShares {
     }
 
     pub async fn clear_rewarded_shares(
-        db: impl sqlx::PgExecutor<'_>,
+        tx: &mut sqlx::Transaction<'_, sqlx::Postgres>,
         period_end: DateTime<Utc>,
     ) -> Result<(), sqlx::Error> {
-        sqlx::query(
-            "delete from gateway_shares where reward_timestamp <= $1;
-                  delete from gateway_dc_shares where reward_timestamp <= $1",
-        )
-        .bind(period_end)
-        .execute(db)
-        .await
-        .map(|_| ())
+        sqlx::query("delete from gateway_shares where reward_timestamp <= $1")
+            .bind(period_end)
+            .execute(&mut *tx)
+            .await
+            .map(|_| ())?;
+
+        sqlx::query("delete from gateway_dc_shares where reward_timestamp <= $1")
+            .bind(period_end)
+            .execute(&mut *tx)
+            .await
+            .map(|_| ())
```

Comment:
Personally, I would do the following: 
```suggestion
        let _ = sqlx::query("delete from gateway_shares where reward_timestamp <= $1")
            .bind(period_end)
            .execute(&mut *tx)
            .await?;

        let _ = sqlx::query("delete from gateway_dc_shares where reward_timestamp <= $1")
            .bind(period_end)
            .execute(&mut *tx)
            .await?;

        Ok(())
```